exercise_solutions.tex

```diff
@@ -1441,6 +1441,245 @@ \subsection*{Solution to \cref{ex:acconn}}
 But also by \cref{ex:connectivity-inductively}, each $Y(x)$ is $(-1)$-connected and all its path types are $n$-connected.
 Applying function extensionality to characterize the path types of $\prd{x:X} Y(x)$, the claim follows from the $(-1)$-connected and $n$-connected axioms of choice.
 
+
+\section*{Exercises from \cref{cha:homotopy}}
+
+\subsection*{Solution to \cref{ex:homotopy-groups-resp-prod}}
+Remember that the fundamental group of a pointed space is defined as the
+$0$-truncation of the loop space of that space.
+If we prove the equivalence without applying $\trunc 0{\blank}$ to both sides
+we are done.  We use induction on the natural numbers, with base case $1$.
+
+Suppose we have two pointed types $(A,a)$ and $(B,b)$.
+For the base case, we have $\eqv{\Omega(A \times B)}{\Omega(A) \times \Omega(B)}$
+because it is, by definition, the equivalence of pointed types:
+
+\[\eqv{ \Parens{(a,b)=(a,b), \refl{(a,b)}} }{\Parens{(a=a \times b=b), (\refl{a}, \refl{b})}}.\]
+
+By Univalence Axiom this is the same as proving the equality.
+By the characterization of \cref{thm:path-sigma} (and using Univalence) we see
+that it suffices to give a:
+\[p:\eqv{\Parens{(a,b)=(a,b)}}{\Parens{(a=a)\times (b=b)}}\]
+
+And to show that transporting $\refl{(a,b)}$ along $\ua(p)$ is equal to
+$(\refl{a}, \refl{b})$.  To construct $p$ just use \cref{thm:path-prod}. To
+show that transport respects the equality we use
+the propositional computation rule in \cref{axiom:univalence} and the definition
+of our function $p$, that is just an application of the projections and
+functoriality (see \cref{eq:path-prod}).
+
+For the inductive step we use the inductive hypothesis to get an equivalence:
+$\eqv{\Omega^n(A \times B)}{\Omega^n(A) \times \Omega^n(B)}$.
+Then we apply $\Omega(\blank)$ on both sides and use exactly the same
+propositions we used in the base
+case. This settles the inductive case, because of the inductive definition of
+$\Omega^{\suc(n)}(\blank)$.
+
+
+\subsection*{Solution to \cref{ex:contr-infinity-sphere-colim}}
+
+To solve this exercise we must first define the spheres as a type family
+$\Sn^{\blank} : \nat \to \UU$.
+Then we must define the inclusions that appear the in the diagram:
+\[ \Sn^0 \to \Sn^1 \to \Sn^2 \to\cdots \]
+
+Then we will be able to define the colimit as
+a higher inductive type.
+
+So, by induction on the natural numbers, we define $\Sn^{\blank} : \nat \to \UU$
+with the base case being the two point type $\bool$, and the inductive case
+being the iterated application of the suspension.
+
+Now we define the inclusions $i_n : \Sn^n \to \Sn^{n+1}$.
+For the base case we have to give a function $i_0 : \bool \to \susp \bool$.
+This is easy: send one point to $\north$ and the other to $\south$.
```

Comment:
That's true. In fact, by starting at 0 one has to repeat part of the argument. I prefer to start at 0 only because I like the family S^- to be defined over the natural numbers. But I can change it if it is not elegant to repeat part of an argument.